tests/ui/impl-trait/impl_fn_associativity.rs

```diff
@@ -7,6 +6,7 @@ fn f_debug() -> impl Fn() -> impl Debug {
 }
 
 fn ff_debug() -> impl Fn() -> impl Fn() -> impl Debug {
+    //~^ ERROR: expected a `Fn()` closure, found `()` [E0277]
```

Comment:
I'd need to properly check what's going on here but at a glance this regression seems wrong?